R/xcmsSet.R

```diff
@@ -253,6 +253,18 @@ split.xcmsSet <- function(x, f, drop = TRUE, ...) {
     lcsets <- vector("list", length(levels(f)))
     names(lcsets) <- levels(f)
 
+    ## get the phenoData and all other parameters we want to pass
+    ## down to the splitted objects
+    pd <- phenoData(x)
+    dataCor <- x@dataCorrection
+    suppressWarnings(
```

Comment:
I've added the mslevel and scanrange slots in a previous commit, thus, they are not present in "older" xcmsSet objects (like the ones from the faahKO package). When the methods mslevel and scanrange are called on such objects a warning is generated stating that the slots are not available and returning the standard values for these slots.